internal/persistence/postgres/pool.go

```diff
@@ -0,0 +1,154 @@
+package postgres
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"go.uber.org/zap"
+)
+
+var pools = map[string]PoolOpenFunc{}
+
+// Pool is the interface of the pgxpool.Pool type, used by the postgres
+// persistence layer to query the database. A custom pool type can be
+// registered using `RegisterPool` if it implements this interface.
+type Pool interface {
+	Close()
+	Acquire(ctx context.Context) (*pgxpool.Conn, error)
+	AcquireFunc(ctx context.Context, f func(*pgxpool.Conn) error) error
+	AcquireAllIdle(ctx context.Context) []*pgxpool.Conn
+	Config() *pgxpool.Config
+	Stat() *pgxpool.Stat
+	Exec(ctx context.Context, sql string, arguments ...interface{}) (pgconn.CommandTag, error)
+	Query(ctx context.Context, sql string, args ...interface{}) (pgx.Rows, error)
+	QueryRow(ctx context.Context, sql string, args ...interface{}) pgx.Row
+	QueryFunc(ctx context.Context, sql string, args []interface{}, scans []interface{},
+		f func(pgx.QueryFuncRow) error) (pgconn.CommandTag, error)
+	SendBatch(ctx context.Context, b *pgx.Batch) pgx.BatchResults
+	Begin(ctx context.Context) (pgx.Tx, error)
+	BeginTx(ctx context.Context, txOptions pgx.TxOptions) (pgx.Tx, error)
+	BeginFunc(ctx context.Context, f func(pgx.Tx) error) error
+	BeginTxFunc(ctx context.Context, txOptions pgx.TxOptions, f func(pgx.Tx) error) error
+	CopyFrom(ctx context.Context, tableName pgx.Identifier, columnNames []string, rowSrc pgx.CopyFromSource) (int64, error)
+	Ping(ctx context.Context) error
+}
+
+// PoolOpenFunc defines a function that can instantiate a Pool instance with the given options.
+type PoolOpenFunc func(opts Opts, logger *zap.Logger) (Pool, error)
+
+type PoolOpts struct {
+	// Optional The name of the postgres pool implementation to use.
+	// By default, only `pgx` is available.
+	Name string
+
+	// MaxConns is the maximum size of the pool. Default to `persistence.DefaultMaxConn`.
```

Comment:
Currently, this is a bit falsy. The default are only set in the config yaml stuff...